pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -224,6 +217,31 @@ func (pv *Provisioner) cloneVirtualMachine(s *SessionContext, cluster *clusterv1
 		return fmt.Errorf("error fetching virtual machine or template properties: %s", err)
 	}
 
+	host, err := src.HostSystem(ctx)
+	if err != nil {
+		return fmt.Errorf("HostSystem failed. err=%s", err)
+	}
+
+	pool, err := s.finder.ResourcePoolOrDefault(ctx, machineConfig.MachineSpec.ResourcePool)
```

Comment:
Can you see if we can make the rp optional? if the user does not specify one then things should work too.

Reply:
Looking into this...

Reply:
Implemented and testing now...